src/plots/cartesian/set_convert.js

```diff
@@ -565,6 +673,141 @@ module.exports = function setConvert(ax, fullLayout) {
+                    isNewBreak = false;
+                }
+            }
+            if(isNewBreak) {
+                breaksOut.push({min: min, max: max});
+            }
+        };
+
+        for(i = 0; i < breaksIn.length; i++) {
+            var brk = breaksIn[i];
+
+            if(brk.enabled) {
+                var op = brk.operation;
+                var op0 = op.charAt(0);
+                var op1 = op.charAt(1);
+
+                if(brk.bounds) {
+                    if(brk.pattern) {
+                        bnds = Lib.simpleMap(brk.bounds, cleanNumber);
+                        if(bnds[0] === bnds[1] && op === '()') continue;
+
+                        // r0 value as date
+                        var r0Date = new Date(r0);
+                        // r0 value for break pattern
+                        var r0Pattern;
+                        // delta between r0 and first break in break pattern values
+                        var r0PatternDelta;
+                        // delta between break bounds in ms
+                        var bndDelta;
+                        // step in ms between breaks
+                        var step;
+                        // tracker to position bounds
+                        var t;
+
+                        switch(brk.pattern) {
+                            case '%w':
+                                b0 = bnds[0] + (op0 === '(' ? 1 : 0);
+                                b1 = bnds[1];
+                                r0Pattern = r0Date.getUTCDay();
+                                r0PatternDelta = b0 - r0Pattern;
+                                bndDelta = (b1 >= b0 ? b1 - b0 : (b1 + 7) - b0) * ONEDAY;
+                                if(op1 === ']') bndDelta += ONEDAY;
+                                step = 7 * ONEDAY;
+
+                                t = r0 + r0PatternDelta * ONEDAY -
+                                    r0Date.getUTCHours() * ONEHOUR -
+                                    r0Date.getUTCMinutes() * ONEMIN -
+                                    r0Date.getUTCSeconds() * ONESEC -
+                                    r0Date.getUTCMilliseconds();
+                                break;
+                            case '%H':
+                                b0 = bnds[0];
+                                b1 = bnds[1];
+                                r0Pattern = r0Date.getUTCHours();
+                                r0PatternDelta = b0 - r0Pattern;
+                                bndDelta = (b1 >= b0 ? b1 - b0 : (b1 + 24) - b0) * ONEHOUR;
+                                step = ONEDAY;
+
+                                t = r0 + r0PatternDelta * ONEHOUR -
+                                    r0Date.getUTCMinutes() * ONEMIN -
+                                    r0Date.getUTCSeconds() * ONESEC -
+                                    r0Date.getUTCMilliseconds();
+                                break;
+                        }
+
+                        while(t <= r1) {
```

Comment:
Could we add a finite test for `t` before entring this loop?
Is there any scenario (e.g. on reversed axes) in which step could be negative?